BLSQ/openhexa-app
# Notebooks in workspaces

This PR integrates the notebooks feature within workspaces. It allows users to launch one jupyter server by workspace. These workspace-specific notebooks servers will be launched with the credentials / env variables of the workspace only.

## Changes

- Added a GraphQL mutation that launches jupyter servers using the jupyterhub REST API (and named servers)
- Added a standard HTTP endpoint the the Jupyterhub spawner can call to get the workspace credentials

## How/what to test

This can be tested locally - you need to run `openhexa-notebooks` using Docker.

- Run the `launchNotebookServer` mutation
- Navigate to the URL returned by the mutation
- Make sure that the notebook server runs fine and has workspace-specific credentials

## Screenshots / screencast

If relevant, please add some screenshots or a short video.
